src/ng/directive/ngEventDirs.js

```diff
@@ -65,6 +65,8 @@ forEach(
               };
               if (forceAsyncEvents[eventName] && $rootScope.$$phase) {
                 scope.$evalAsync(callback);
+              } else if ($rootScope.$$phase) {
```

Comment:
In order to avoid double checking `$rootScope.$$phase` and simplify, I might change the whole block to:

```js
if (!$rootScope.$$phase) {
  scope.$apply(callback);
} else if (forceAsyncEvents[eventName]) {
  scope.$evalAsync(callback);
} else {
  callback();
}
```